File: Linguagem de Programação I/Extras/Beecrowd_Extra/1212.ts
import * as readline from 'readline';

let rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input: string[] = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    for (let i = 0; i < input.length; i++) {
        let a: string[] = input[i].split(' ')[0].split('');
        let b: string[] = input[i].split(' ')[1].split('');
        let carry = 0;
        let c = 0;
        if (a.join('') === '0' && b.join('') === '0') {
            break;
        }

        while (a.length < b.length) {
            a.unshift('0');
        }
        while (b.length < a.length) {
            b.unshift('0');
        }

        for (let i = a.length - 1; i >= 0; i--) {
            if ((parseInt(a[i]) + parseInt(b[i]) > 9) || (parseInt(a[i]) + parseInt(b[i]) + c > 9)) {
                carry += 1;
                c = 1;
            } else {
                c = 0;
            }
        }

        if (carry === 0) {
            console.log("No carry operation.");
        } else if (carry === 1) {
            console.log("1 carry operation.");
        } else {
            console.log(carry + " carry operations.");
        }
    }
});

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1212.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    loop {
        let line = lines.next().unwrap().unwrap();
        let mut parts = line.split_whitespace();
        let mut a: Vec<char> = parts.next().unwrap().chars().collect();
        let mut b: Vec<char> = parts.next().unwrap().chars().collect();
        let mut carry = 0;
        let mut c = 0;
        if a.iter().collect::<String>() == "0" && b.iter().collect::<String>() == "0" {
            break;
        }

        while a.len() < b.len() {
            a.insert(0, '0');
        }
        while b.len() < a.len() {
            b.insert(0, '0');
        }

        for i in (0..a.len()).rev() {
            if ((a[i].to_digit(10).unwrap() + b[i].to_digit(10).unwrap() > 9) ||
                (a[i].to_digit(10).unwrap() + b[i].to_digit(10).unwrap() + c > 9)) {
                carry += 1;
                c = 1;
            } else {
                c = 0;
            }
        }

        if carry == 0 {
            println!("No carry operation.");
        } else if carry == 1 {
            println!("1 carry operation.");
        } else {
            println!("{} carry operations.", carry);
        }
    }
}